Current plugin version features a mismatch between Quarkus versions

## Bug Report

The problem is that the generated POM uses a Quarkus version that differs from the one provided by the SDK Quarkus extension. This results in errors similar to:

```java
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalStateException: Hot deployment of the application is not supported when updating the Quarkus version. The application needs to be stopped and dev mode started up again
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:138)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: java.lang.IllegalStateException: Hot deployment of the application is not supported when updating the Quarkus version. The application needs to be stopped and dev mode started up again
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:153)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:106)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:132)
	... 1 more
Caused by: java.lang.IllegalStateException: Hot deployment of the application is not supported when updating the Quarkus version. The application needs to be stopped and dev mode started up again
	at io.quarkus.deployment.dev.IsolatedDevModeMain.close(IsolatedDevModeMain.java:337)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:482)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:66)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:150)
	... 3 more
```

See also: https://github.com/quarkiverse/quarkus-operator-sdk/issues/248

`$ operator-sdk version`
1.18.0

#### Possible Solution

This requires a plugin update to fix. Users affected by the problem can simply change the Quarkus version specified in the generated pom at line 19 to use version `2.6.3.Final` instead of `2.7.0.Final`. 

#### Additional context

We should probably add a test that scaffolds a project and verifies that it builds properly as part of CI.

--- a/pkg/quarkus/v1alpha/scaffolds/internal/templates/pomxml.go
+++ b/pkg/quarkus/v1alpha/scaffolds/internal/templates/pomxml.go
@@ -57,8 +57,8 @@ const pomxmlTemplate = `<?xml version="1.0" encoding="UTF-8"?>
     <maven.compiler.target>11</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <quarkus-sdk.version>3.0.2</quarkus-sdk.version>
-    <quarkus.version>2.7.0.Final</quarkus.version>
+    <quarkus-sdk.version>3.0.4</quarkus-sdk.version>
+    <quarkus.version>2.7.3.Final</quarkus.version>
   </properties>
 
   <dependencyManagement>

